src/hierarchies/likelihoods/abstract_likelihood.h

```diff
@@ -0,0 +1,187 @@
+#ifndef BAYESMIX_HIERARCHIES_LIKELIHOODS_ABSTRACT_LIKELIHOOD_H_
+#define BAYESMIX_HIERARCHIES_LIKELIHOODS_ABSTRACT_LIKELIHOOD_H_
+
+#include <google/protobuf/message.h>
+
+#include <memory>
+#include <stan/math/prim.hpp>
+#include <stan/math/rev.hpp>
+
+#include "algorithm_state.pb.h"
+
+class AbstractLikelihood {
+ public:
+  //! Default destructor
+  virtual ~AbstractLikelihood() = default;
+
+  //! Returns an independent, data-less copy of this object. Implemented in
+  //! BaseLikelihood
```

Comment:
Again, I would remove this "Implemented in BaseLikelihood", also in other points of this file

Reply:
agreed. I get why you put these comments here, but IMO it would be better to have a well written Readme file in the src/hierarchies/likelihoods folder that explains exactly which methods one needs to implement in an actual Likelihood.

Reply:
I have removed all these stuff. I will leave this open for the DOCS stuff that @mberaha is suggesting.